Linking customer name from transaction detail page to filtered list page

shendy-a8c opened the pull request:
Fixes #1154

#### Changes proposed in this Pull Request

* Creates a new component: CustomerLink to display link on transaction detail page back to transaction list page filtered by the customer's name.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Run `npm start`.
* Open the [transaction list page](http://localhost:8082/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions).
* Open one of the transaction's detail page by clicking on of the transaction row.
* With branch `master`, customer name is displayed as a text.
* With this branch, customer name is displayed as a link.
* Click on the customer name.
* Confirm you are back on the transaction list page but filtered by customer's name.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

<!--
Please read P7bbVw-3ww-p2 before opening the PR, assign the correct status to it, __and make sure that the related issue uses the Has PR status!__.
-->

jrodger commented:
Looks great!

Running through the tests works as expected. When I click the link I arrive at the same URL that I end up at when searching manually, so this seems like a robust approach to me.

Just needs a rebase on `trunk` and then is ready to merge.

shendy-a8c commented:
[x] Rebased.
[x] Added a change line to `readme.txt` as I only added to `changelog.txt`.

Investigating what's wrong with [the checks](https://github.com/Automattic/woocommerce-payments/pull/1163/checks) as not all of them are green.

shendy-a8c commented:
Checks on WooCommerce beta version (specifically 4.9.0-rc.1) failed [#](https://github.com/Automattic/woocommerce-payments/runs/1667501448) [#](https://github.com/Automattic/woocommerce-payments/runs/1667501462).

However, I read ['allowed to fail' here](https://github.com/Automattic/woocommerce-payments/blob/update/link-customer-name-on-tx-detail/.github/workflows/compatibility.yml#L43), so merging this PR.
